### DISCUSSION TITLE
Priorities in Documentation Topics

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#134] [Design & Development]
I think we need a conversation on priorities in documentation we need to develop in the near future for our initial release.   It is important to have sufficient documentation when we announce this that people know how to assemble the package, get started, and understand the key concepts the package uses.   We have some, but not enough.  Here are some suggestions in my priority order.   Please add and suggest alternative ordering:
1. A more complete document on installing MsPASS and setting it up for docker and singularity
2. A companion tutorial (jupyter) to 1 with a title like "Getting Started".   I think a good approach would be to 
    download some data with IRIS web services and build a small simple dataset of two good teleseimic events 
    recorded by the USArray, pulling stationxml data, and quakeml files for the two events.   This differs 
    from our monster data set in that it should be feasible to do on the fly.   Once the data is assembled we 
    could have them run a serial and parallel version of the same workflow and plot the results.  The workflow
    could be a work in progress for a while but that basic content seems sensible as a "getting started".
3. More documentation on using MongoDB and the Database and Client classes.   We need something more 
    introductory than what I wrote before.   Probably need an "Introduction to Database Concepts" or 
    some such title and a related jupyter tutorial.   Unclear if we should recycle existing material for some 
    part of this and put the more detailed material into a reference manual document.  In short, I think what
    I wrote to now is inadequate.   
4. We desperately need an introduction to the Parallel API and a related jupyter tutorial.   I cannot yet 
     write this as my knowledge base is inadequate.  
5.  We need a "Writing a MsPASS Module" section for the reference manual or as an "Advanced Topic".
6. We need a "Adding a Custom Schema" section for the reference manual or as an "Advanced Topic".
7.  We need a more extensive section on containers in general and how you configure the docker
     and singularity containers.   If you got died of covid tomorrow this development would die because 
     I don't think I could reconstruct what you've done.   A classic reason why documentation is 
     essential.
8.  We need a more comprehensive discussion of the elog mechanism and a related tutorial.
9.  Probably should have a User Manual section on graphics to go with the existing tutorial.   The priority of 
     this would be higher but I think the graphics system will evolve as we develop some of these other 
     tutorials. 
10. When we finish the global history design we need a set of pages on the global and object level history
      use.   As with most of these a parallel jupyter tutorial would be good.

That is now a long list and without doubt is missing something.  Look forward to hearing your opinions.

**Comment by wangyinz** [comment]
Thanks for listing these items, I think this pretty much covers everything. Item 1 is definitely the most important, but I was to write it later because the container part of the package actually needs some fundamental change. We need to also support dask, and add script to handle different options to deploy distributed database and processors (Spark and Dask). Current setup for container is not quite user friendly and lacks features.

So, maybe we can focus on the rest for now. I can see that the database API and parallel API are pretty much stable now. Maybe we should start from there? After that, we can work on the getting started guide, and then the rest (which should are less important for our initial release).

**Reply by pavlis** [reply]
Good to get your perspective.   I see the point about doing the database and parallel api soon.   However, I think I will take on the "Getting Started" guide today.   I don' think I can do the parallel tutorial justice and I have a few things to learn yet about the database api.  I think writing getting started will educate me more on holes in my understanding that will allow me to take on the database api next.
